distribution.md

```diff
@@ -43,29 +44,219 @@ Each buildpack layer blob MUST contain a single buildpack at the following file
+
+A platform supporting builders SHOULD allow users to configure the build environment for a provided builder.
+
+#### General Requirements
+
+The builder image MUST contain an implementation of the [lifecycle](#lifecycle), and a [build-time](#build-image) environment, and MAY contain [buildpacks](#buildpackage). Platforms SHOULD use builders to ease the build process.
+
+#### Filesystem
+
+A builder MUST have the following directories/files:
+
+* `/cnb/order.toml` &rarr; As defined in the [platform specification](https://github.com/buildpacks/spec/blob/main/platform.md#ordertoml-toml).
+
+#### Environment Variables
+
+A builder MUST be an extension of a Build Image, and MUST retain all the specified environment variables set on the original build image, as specified in the Build Image specifications.
+
+#### Labels
+
+A builder MUST be an extension of a Build Image, and MUST retain all the specified Labels set on the original build image, as specified in the Build Image specifications.
+
+A builder image MUST contain an implementation of the [lifecycle](#lifecycle), and MUST retain all the specified Labels set on the original Lifecycle image, as specified in the lifecycle distribution specifications.
+
+A builder image MAY contain buildpacks, and MAY retain all the specified Labels set on the original buildpackage, as specified in the [buildpackage](#buildpackage) specification with the following exceptions:
 
-For each listed stack, all associated buildpacks MUST be a candidate for detection when the entrypoint buildpack ID and version are selected.
+- `io.buildpacks.buildpack.metadata` MUST not be set.
+- `io.buildpacks.buildpack.layers`  on the builder SHOULD be a merged version based on all buildpackages combined and thereby have of all packaged buildpacks represented.
+
+In addition to all inherited labels above, the following labels MUST be set in the builder environment (through the image config's `Labels` field):
+
+| Label                            | Description                                                   |
+| -------------------------------- | ------------------------------------------------------------- |
+| `io.buildpacks.builder.metadata` | A JSON object representing builder metadata.                  |
+| `io.buildpacks.buildpack.order`  | A JSON object representation of the `/cnb/order.toml` file.   |
+
+##### `io.buildpacks.builder.metadata` (JSON)
+
+```json
+{
+  "description": "<string>",
+  "createdBy": {
+    "name": "<string>",
+    "version": "<string>",
+  }
+}
+```
+
+Where:
+
+- `description` (optional) is a human readable description of the builder.
+- `createdBy` (optional) is information about the tool that created the builder.
+  - `name` (optional) is the name of the tool that created the builder.
+  - `version` (optional) is the version of the tool that created the builder.
+
+## Data Format
+
+### Files
+
+#### `builder.toml` (TOML)
```

Comment:
Do we reference how this file is used anywhere in this spec?

It's a little different from the builder image itself, because it's actually input to the tool that creates the image.

Reply:
fair - so imo it's one of the more useful parts of the spec - do we need to say something like "a platform implementation MUST be able to produce a valid builder from a valid builder.toml input file"?

Reply:
I added something like that under Builder.General  - lmk wdyt?